addons/disposable/functions/fnc_replaceATWeapon.sqf

```diff
@@ -34,33 +34,24 @@ _unit selectWeapon _replacementTube;
     if (_x != "") then {_unit addSecondaryWeaponItem _x};
 } count _items;
 
-
-// AI - Remove the ai's missle launcher tube after the missle has exploded
-if !([_unit] call EFUNC(common,isPlayer)) then {
-    [{
-        params ["_args","_idPFH"];
-        _args params ["_unit", "_tube", "_projectile"];
-
-        //don't do anything until projectile is null (exploded/max range)
-        if (isNull _projectile) then {
-            //Remove PFEH:
-            [_idPFH] call CBA_fnc_removePerFrameHandler;
-
-            //If (tube is dropped) OR (is dead) OR (is player) just exit
-            if (secondaryWeapon _unit != _tube || {!alive _unit} || {[_unit] call EFUNC(common,isPlayer)}) exitWith {};
-
-            //private  _items = secondaryWeaponItems _unit;
-            private _container = createVehicle ["GroundWeaponHolder", position _unit, [], 0, "CAN_COLLIDE"];
-            _container setPosAsl (getPosAsl _unit);
-            _container addWeaponCargoGlobal [_tube, 1];
-
-            //This will duplicate attachements, because we will be adding a weapon that may already have attachments on it
-            //We either need a way to add a clean weapon, or a way to add a fully configured weapon to a container:
-            // {
-                // if (_x != "") then {_container addItemCargoGlobal [_x, 1];};
-            // } forEach _items;
-
-            _unit removeWeaponGlobal _tube;
-        };
-    }, 1, [_unit, _replacementTube, _projectile]] call CBA_fnc_addPerFrameHandler;
-};
+[{
+    params ["_args", "_idPFH"];
+    _args params ["_unit", "_replacementTube", "_projectile"];
+
+    //Wait until the unit switches weapon.
+    if (currentWeapon _unit != _replacementTube) then {
+        //Remove PFEH:
+        [_idPFH] call CBA_fnc_removePerFrameHandler;
+
+        //If (tube is dropped) OR (is dead) just exit
+        if (secondaryWeapon _unit != _replacementTube || {!alive _unit}) exitWith {};
+
+        //Remove the weapon and "throw" it on the ground.
+        _unit removeWeaponGlobal _replacementTube;
+        private _container = createVehicle ["WeaponHolderSimulated", _unit modelToWorld ((_unit selectionPosition ["rightshoulder", "Memory"]) vectorAdd [0, 0.2, 0.1]), [], 0, "CAN_COLLIDE"];
```

Comment:
`createVehicle` needs ATL pos (except for boat or amphibious vehicles, where it needs AGL), but m2w reports AGL.

This breaks on aircraft carriers.

Reply:
@commy2 I appreciate the code review, but now I'm confused as to the status of this.

1. The PR itself is still closed, so the code here is stale by at least two weeks as can be seen [here](https://github.com/CoxLedere/ACE3/commits/DisposableExpansion)
2. Your comment from [here](https://github.com/acemod/ACE3/pull/5422#issuecomment-482780446) seems incredibly damning on this ever getting into ACE.

Ultimately, I appreciate you pointing out the problem here, I'm just confused why you're spending time on something that is effectively "dead"

Reply:
I am doing something similar in CBA and these are notes for me.